pythclient/hermes.py

```diff
@@ -0,0 +1,173 @@
+        """
+        Queries the Hermes https endpoint for a list of the IDs of all Pyth price feeds.
+        """
+
+        url = os.path.join(self.endpoint, "price_feed_ids")
+        
+        client = httpx.AsyncClient()
+
+        data = (await client.get(url)).json()
+
+        return data
+
+    def add_feed_ids(self, feed_ids: list[str]):
+        self.feed_ids += feed_ids
+        self.feed_ids = list(set(self.feed_ids))
+        self.pending_feed_ids += feed_ids
+
+    @staticmethod
+    def extract_price_feed(data: dict) -> PriceFeed:
+        """
+        Extracts a PriceFeed object from the JSON response from Hermes.
+        """
+        price = Price.from_dict(data["price"])
+        ema_price = Price.from_dict(data["ema_price"])
+        vaa = data["vaa"]
+        price_feed = {
+            "feed_id": data["id"],
+            "price": price,
+            "ema_price": ema_price,
+            "vaa": vaa,
+        }
+        return price_feed
+
+    async def get_pyth_prices_latest(self, feedIds: list[str]) -> list[PriceFeed]:
+        """
+        Queries the Hermes https endpoint for the latest price feeds for a list of Pyth feed IDs.
+        """
+        url = os.path.join(self.endpoint, "latest_price_feeds?")
+        params = {"ids[]": feedIds, "binary": "true"}
+
+        data = (await self.client.get(url, params=params)).json()
+
+        results = []
+        for res in data:
+            price_feed = self.extract_price_feed(res)
+            results.append(price_feed)
+
+        return results
+
+    async def get_pyth_price_at_time(self, feed_id: str, timestamp: int) -> PriceFeed:
+        """
+        Queries the Hermes https endpoint for the price feed for a Pyth feed ID at a given timestamp.
+        """
+        url = os.path.join(self.endpoint, "get_price_feed")
+        params = {"id": feed_id, "publish_time": timestamp, "binary": "true"}
+
+        data = (await self.client.get(url, params=params)).json()
+
+        price_feed = self.extract_price_feed(data)
+
+        return price_feed
+
+    async def get_all_prices(self) -> dict[str, PriceFeed]:
+        """
+        Queries the Hermes http endpoint for the latest price feeds for all feed IDs in the class object.
+
+        There are limitations on the number of feed IDs that can be queried at once, so this function queries the feed IDs in batches.
+        """
+        pyth_prices_latest = []
+        i = 0
+        batch_size = 100
+        while len(self.feed_ids[i : i + batch_size]) > 0:
+            pyth_prices_latest += await self.get_pyth_prices_latest(
+                self.feed_ids[i : i + batch_size]
+            )
+            i += batch_size
+
+        return dict([(feed['feed_id'], feed) for feed in pyth_prices_latest])
+
+    async def ws_pyth_prices(self):
+        """
+        Opens a websocket connection to Hermes for latest prices for all feed IDs in the class object.
+        """
+        import json
+
+        import websockets
```

Comment:
is there a specific reason to do this? otherwise I recommend to import it at top-level to improve readability and maintainability